SoftwareCo/SoftwareCo/Tracker/managers/HashManager.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using Newtonsoft.Json;
+using Sodium;
+
+namespace SoftwareCo
+{
+    public class HashManager
+    {
+        public static String HashValue(string value, string dataType)
+        {
+            if (!CacheManager.HasJwt() || string.IsNullOrEmpty(value))
+            {
+                return "";
+            }
+
+            try
+            {
+                byte[] key = GenericHash.GenerateKey(); // 64 byte key
+                string hashedValue = Utilities.BinaryToHex(GenericHash.Hash(value, key, 128));
+                if (CacheManager.HasCachedValue(dataType, hashedValue))
+                {
+                    // doesn't exist yet, encrypt it
+                    EncryptValue(value, hashedValue, dataType);
+
+                    // refresh the cache
+                    PopulateHashValues();
+                }
+                return hashedValue;
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("SwdcVsTracker - Hash value error: {0}", e.Message);
+            }
+            return "";
+        }
+
+        private static async void EncryptValue(string value, string hashedValue, string dataType)
+        {
+            UserEncryptedData data = new UserEncryptedData();
```

Comment:
do we really need the UserEncryptedData object?   Can we just put the data in the call to PostAsync directly?